### ISSUE TITLE
Incorrect KaTex markdown rendering with `\[` sequence

### ISSUE THREAD

**Issue by silverwind** [go-gitea/gitea#24006]
### Description

Given this markdown:

````markdown
-   \[CSS] - Nested atRule's `@media` or `@supports` now properly are replaced with unique identifiers by [@&#8203;noreiller](https://github.com/noreiller) in https://github.com/webpack/webpack/pull/15812
````

It renders

<img width="769" alt="image" src="https://user-images.githubusercontent.com/115237/230737099-377b137c-02b6-49a4-a1b4-fd9dbc5cc244.png">

It should render like GitHub

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/115237/230737118-1ab6e96b-8bea-4cd3-8e2c-8930bf11a8f6.png">

### Gitea Version

main

### Can you reproduce the bug on the Gitea demo site?

Yes

**Comment by silverwind** [comment]
So I think the bug is in the backend where it detects `\[CSS` as math content. I think we need to be more strict and require the `$$` wrapping around math, like below:

```
$$\left( \sum_{k=1}^n a_k b_k \right)^2 \leq \left( \sum_{k=1}^n a_k^2 \right) \left( \sum_{k=1}^n b_k^2 \right)$$

**Comment by nschloe** [comment]
Gitea was one of the first services to offer math support using a `\(...\)`/`\[...\]` syntax. Unfortunatly, it has not been adopted by any other platform, perhaps also because of the fact that it collides with standard Markdown syntax. (More about this in https://github.com/go-gitea/gitea/issues/27645.)

Right now, GitLab, GitHub, and Pandoc support two syntaxes:

- `$...$`/`$$...$$` (traditional TeX),
- `` $`...`$ ``/ ```` ```math... ``` ```` (originally introduced by GitLab in 2016).

[In a survey across GitHub, GitLab, Gitea, and Pandoc](https://nschloe.github.io/2023/11/16/the-state-of-math-in-markdown.html#comparison-table), I've found that the `$`-syntax is error prone, also because it's non-native to Markdown. The backticked syntax (not supported by Gitea right now) is a great practical alternative.

For full compatibility with other platform, perhaps Gitea can think about sunsetting the bracket syntax for math, and add the backticked syntax. This would also fix this bug here.

**Comment by wxiaoguang** [comment]
Will be fixed by #32831

**Comment by wxiaoguang** [comment]
The `\[ ... ]` has been fixed in #32831, but the `\[ ... \]` is still problematic, I have left a TODO in code, we should drop this syntax in the future by a config option.

**Comment by wxiaoguang** [comment]
The complete fix is  Fix markdown render behaviors #34122
